internal/uvm/scsi.go

```diff
@@ -194,20 +232,58 @@ func (uvm *UtilityVM) AddSCSI(ctx context.Context, hostPath string, uvmPath stri
 // `uvmPath` is optional if a guest mount is not requested.
 //
 // `readOnly` set to `true` if the physical disk should be attached read only.
-func (uvm *UtilityVM) AddSCSIPhysicalDisk(ctx context.Context, hostPath, uvmPath string, readOnly bool) (*SCSIMount, error) {
-	return uvm.addSCSIActual(ctx, hostPath, uvmPath, "PassThru", false, readOnly)
+func (uvm *UtilityVM) AddSCSIPhysicalDisk(ctx context.Context, hostPath, uvmPath string, readOnly bool) (sm *SCSIMount, err error) {
+	sm, existed, err := uvm.allocateSCSIDevice(ctx, hostPath, uvmPath, VMAccessTypeIndividual)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		if err != nil {
+			uvm.deallocateSCSI(ctx, sm)
+		}
+	}()
+
+	if existed {
+		return sm, nil
+	}
+
+	err = uvm.addSCSIActual(ctx, sm, "PassThru", readOnly)
+	if err != nil {
+		return nil, err
+	}
+	return sm, err
 }
 
 // AddSCSILayer adds a read-only layer disk to a utility VM at the next
 // available location and returns the path in the UVM where the layer was
 // mounted. This function is used by LCOW as an alternate to PMEM for large
 // layers.
-func (uvm *UtilityVM) AddSCSILayer(ctx context.Context, hostPath string) (*SCSIMount, error) {
+//
+// This is a read-only (LCOW) layer VHD. This function `must not` be used for Windows.
+func (uvm *UtilityVM) AddSCSILayer(ctx context.Context, hostPath string) (sm *SCSIMount, err error) {
```

Comment:
I think we should just remove `AddSCSILayer`. The caller should just compute the UVM path itself, and call `AddSCSI`, passing in `VMAccessTypeNoop`.

Reply:
we've been using controller and lun for the uvmPath for layers though. I'm fine with this change so long as no one has an issue with moving away from controller & lun in the layers uvmPath to something more of this form:

`/run/layers/s<uvm layer index>`

Reply:
Maybe we should divide this PR into two:
- Minimal fix to unblock GPU
- Larger refactoring to clean up this code
Right now there is a fair amount of refactoring mixed in with the GPU fix. What do you think?

Reply:
Could we not just move the larger removal of AddSCSILayers into a different PR? Is there something about this PR in particular you think should be removed for minimal fix of gpu scenarios?

Reply:
My thought was if we're doing refactoring in this PR, we should make sure the end result is what we want longer term. However, if you don't have cycles to do more refactoring right now, we can probably take most of this as-is. I'll take another look.